bundles/org.openhab.binding.hive/src/main/java/org/openhab/binding/hive/internal/client/dto/FeatureDtoBase.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hive.internal.client.dto;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.hive.internal.client.FeatureType;
+
+/**
+ *
+ *
+ * @author Ross Brown - Initial contribution
+ */
+@NonNullByDefault
+public abstract class FeatureDtoBase {
+    public @Nullable FeatureAttributeDto<FeatureType> featureType;
```

Comment:
Does the hive api really utilize all of the fields in `FeatureAttributeDto` here?

Reply:
The API can be a little weird with which fields it uses for specific attributes so I decided to use one DTO with all the fields instead of multiple with different subsets of fields.

Almost all attributes use these:
- `reportedValue`
- `displayValue`
- `reportChangedTime`
- `reportReceivedTime`

For "settable" attributes you can post `targetValue` and will get these in addition.
- `targetValue`
- `targetSetTime`
- `targetExpiryTime`
- `targetSetTXId`
- `propertyStatus`

These will disappear if the update succeeds (`reportedValue` changes to the posted `targetValue`) or stick around if the update fails (with `propertyStatus` = "FAILED")

Weirdly `featureType` always has a `targetValue` (even though it cannot be set) and has no `reportXxxTime` fields.

Some nodes with the `physical_device_v1` feature have the `nativeIdentifier` attribute which is normal.
Others have the `hardwareIdentifier` attribute which unusually has no `reportXxxTime` fields.